File: app/src/main/java/com/androidskeleton/module/ui/home/HomePresenter.java
package com.androidskeleton.module.ui.home;

import com.androidskeleton.di.scope.PerActivity;

import javax.inject.Inject;

@PerActivity
public class HomePresenter implements HomeContract.Presenter {
    private HomeContract.View view;

    @Inject
    public HomePresenter() {
    }

    @Override
    public void sayHello(){
        view.greetings("Hello! I am presenter at your service!");
    }

    @Override
    public void takeView(HomeContract.View view) {
        this.view = view;
    }

    @Override
    public void dropView() {
        view = null;
    }
}

File: app/src/main/java/com/androidskeleton/module/ui/home/HomePresenter.kt
package com.androidskeleton.module.ui.home

import com.androidskeleton.di.scope.PerActivity
import com.androidskeleton.module.ui.home.HomeContract.Presenter
import javax.inject.Inject

@PerActivity
class HomePresenter @Inject constructor() : Presenter {
    private var view: HomeContract.View? = null

    override fun sayHello() {
        view?.greetings("Hello! I am presenter at your service!")
    }

    override fun takeView(view: HomeContract.View?) {
        this.view = view
    }

    override fun dropView() {
        view = null
    }
}